core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -558,7 +572,19 @@ private ClientFactoryOptions buildOptions() {
             return ClientFactoryOption.ADDRESS_RESOLVER_GROUP_FACTORY.newValue(addressResolverGroupFactory);
         });
 
-        return ClientFactoryOptions.of(options.values());
+        final ClientFactoryOptions newOptions = ClientFactoryOptions.of(options.values());
+        final long idleTimeoutMillis = newOptions.idleTimeoutMillis();
+        final long pingIntervalMillis = newOptions.pingIntervalMillis();
+        final long minPingIntervalMillis = Math.max(pingIntervalMillis, MIN_PING_INTERVAL_MILLIS);
+        if (idleTimeoutMillis > 0) {
+            if (minPingIntervalMillis >= idleTimeoutMillis) {
+                return ClientFactoryOptions.of(newOptions, ZERO_PING_INTERVAL);
+            }
+            if (minPingIntervalMillis == MIN_PING_INTERVAL_MILLIS) {
```

Comment:
Do we need this?

Reply:
When I left the comment there was no
```java
if (pingIntervalMillis == MIN_PING_INTERVAL_MILLIS) {
    return newOptions;
}
```
this condition, so I left the comment. I guess it's fixed now.

Reply:
Ah... thanks!